src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/SafeHandles.Linux.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.DirectoryServices.Protocols
+{
+    internal sealed class ConnectionHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        internal bool _needDispose = false;
+
+        internal ConnectionHandle()
+            :base(true)
+        {
+            OpenLDAP.ldap_initialize(out IntPtr ldap, null);
+            SetHandle(ldap);
+        }
+
+        internal ConnectionHandle(IntPtr value, bool disposeHandle) : base(true)
```

Comment:
`disposeHandle` here seems to be redundant with the one from the base class... which you're always giving the literal `true`.

If it means "false to never call ReleaseHandle", that's exactly what the base `bool ownsHandle` is for.
https://github.com/dotnet/runtime/blob/5d5c36bb9e887cfe50913840ba96985da113640f/src/coreclr/src/vm/safehandle.cpp#L114-L136

Reply:
So is your suggestion to basically instead of having a field for _needDispose and checking this in ReleaseHandle, I should just call base passing in disposeHandle value and not need to check in ReleaseHandle as it would not get called in the case it was set to false right? Just making sure I understood this comment..

Reply:
Actually we do have one problem with doing this. The one in the base I believe can't be overwritten after creation, and we have some (internal) methods in LdapConnection that can flip the _needsDispose, so I guess the reason why the original design was like this, was so that ReleaseHandle would get called always, and we would only actually dispose if we need to.